cdap-docs/developers-manual/source/advanced/application-logback.rst

```diff
@@ -0,0 +1,43 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _application-logback:
+
+====================
+Application Logbacks
+====================
+
+YARN containers spawned by a CDAP application use the default container logback, ``logback-container.xml``,
+packaged with CDAP. This ``logback.xml`` does log rotation, and depending on the use case, may be sufficient.
+However, it is also possible to specify a custom ``logback.xml`` for a CDAP application.
+The packaged ``logback.xml`` is then used for each container spawned by a program of the application.
+
+Writing and packaging a logback.xml
+===================================
+To specify a ``logback.xml`` for use by an application, all that's required is to package the XML file as a resource
+in the application jar file. The ``pom.xml`` of the application, with the application's ``resource`` directory
+containing the ``logback.xml``, might look like this::
+
+  <?xml version="1.0" encoding="UTF-8"?>
+    <project xmlns="http://maven.apache.org/POM/4.0.0"
+             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+             xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    ...
+
+      <build>
+        <resources>
+          <resource>
+            <directory>resources</directory>
+          </resource>
+        </resources>
+      </build>
+
+    ...
+
+  </project>
+
+CDAP supports standard logback functionality. To write a custom logback, you may
+refer to `LOGBack <http://logback.qos.ch/>`__. It's important to note that when a ``logback.xml`` is
```

Comment:
It's important to note that when a custom ``logback.xml`` is specified for an application, the custom ``logback.xml`` will be used in place of ``logback-container.xml``. The custom ``logback.xml`` needs to have configuration for log rotation and log clean-up, to ensure that long-running containers don't fill up the disk.